src/Test/Inspection/Core.hs

```diff
@@ -102,14 +110,45 @@ pprSlice slice =
 
 -- | Pretty-print two slices, after removing variables occurring in both
 pprSliceDifference :: Slice -> Slice -> SDoc
-pprSliceDifference slice1 slice2 =
-    hang (text "LHS" Outputable.<> colon) 4 (pprSlice slice1') $$
-    hang (text "RHS" Outputable.<> colon) 4 (pprSlice slice2')
+pprSliceDifference slice1 slice2
+    | [(v1,e1)] <- slice1'
+    , [(v2,e2)] <- slice2'
+    = pprSingletonSliceDifference v1 v2 e1 e2
+
+    | otherwise =
+        hang (text "LHS" Outputable.<> colon) 4 (pprSlice slice1') $$
+        hang (text "RHS" Outputable.<> colon) 4 (pprSlice slice2')
   where
     both = S.intersection (S.fromList (map fst slice1)) (S.fromList (map fst slice2))
     slice1' = filter (\(v,_) -> v `S.notMember` both) slice1
     slice2' = filter (\(v,_) -> v `S.notMember` both) slice2
 
+pprSingletonSliceDifference :: Var -> Var -> CoreExpr -> CoreExpr -> SDoc
+pprSingletonSliceDifference v1 v2 e1 e2 =
+    hang (text "In") 4 (ppr $ mkContextExpr (reverse (map snd rn))) $$
+    hang (text "LHS" Outputable.<> colon) 4 (pprSlice [(v1, e1')]) $$
+    hang (text "RHS" Outputable.<> colon) 4 (pprSlice [(v2, e2')])
+  where
+    (e1', e2', rn) = go e1 e2 [] (mkRnEnv2 emptyInScopeSet)
```

Comment:
This `rn` is shadowed by the local one below, can lead to confusion.

And really, this is the context, right? If we add more context things, this would get more compliated? So maybe

```suggestion
    (e1', e2', contxt) = go e1 e2 [] (mkRnEnv2 emptyInScopeSet)
```

Reply:
Oh, I'd just avoid shadowing when it can be confusing (where clause vs. recursive function in that clause). I disabled the warning because I find it annoying when it complains about shadowing imported identifiers 

Not a big deal either way